Canonical mo option

FoleyLab opened the pull request:
This version provides different MO options (canonical HF MO, CQED-HF, CIS natural orbitals) and also provides the option to change the size of the photonic basis using the keyword 'number_of_photons' in the cavity_options dictionary.  Setting 'number_of_photons' : 0 and also setting lambda = [0, 0, 0] will perform ordinary CI calculations effectively in the electronic space only since only the |0> state will be present and it will not contribute anything to the DSE if \lambda = 0,0,0.

Note that    Setting 'number_of_photons' : 1 will include |0> and |1> basis states, number_of_photons' : 2 will include |0>, |1>, |2>, and so on.

There are several items to-do (see the notebook https://github.com/mapol-chem/qed-ci/blob/canonical_mo_option/src/examples/run_helper_PFCI-UPDATED.ipynb for more details)

To-Do:

- [x] Verify that the structure of the PF Hamiltonian is correct for 'number_of_photons' : 2 and larger; the existing tests are based on 'number_of_photons' : 1 and do not break with the updated build, but need to check for larger photonic spaces
- [ ] Investigate the QED-FCI LiH stretch as a function of the size of the photonic basis.  The notebook mentioned above has more details on the geometries and cavity parameters, and has benchmark pCQED results to compare QED-FCI to
- [ ] Investigate the LiH polariton states as a function of coupling strength and different sizes of the photonic basis.

FoleyLab committed:
new branch

FoleyLab committed:
started adding features to compute pf-ci tdm

FoleyLab committed:
some dipole expectation value stuff put into place

FoleyLab committed:
new approach for canonical mos

FoleyLab committed:
function to build MU matrix in CI basis

FoleyLab committed:
tdm seems to work!

FoleyLab committed:
added test for ground-state dipole moment

FoleyLab committed:
Delete test_helperPFCI.cpython-39-pytest-7.1.2.pyc

FoleyLab committed:
Delete h2o_casci-checkpoint.ipynb

FoleyLab committed:
Delete run_mghp-tdm-checkpoint.ipynb

FoleyLab committed:
preliminary results with pCQED

FoleyLab committed:
updated notebook

FoleyLab committed:
some pqed results for lih

FoleyLab committed:
updated data

FoleyLab committed:
updated some helper functions

FoleyLab committed:
added helper methods

FoleyLab committed:
trying to generalize number of photon states for H_PF

FoleyLab committed:
Now I think we can go beyond 1 photon states!

FoleyLab committed:
pr ready

FoleyLab committed:
fixed the G terms

FoleyLab committed:
another helper notebook

FoleyLab committed:
Merge branch 'canonical_mo_option' of https://github.com/mapol-chem/qed-ci into canonical_mo_option

FoleyLab committed:
now building explicit DSE matrix for pCQED

nhv17 reviewed (commented):
the structure of pf hamiltonian has been verified to be correct for h2(sto-3g) up to 4 photon.

FoleyLab committed:
lots of notebooks - need cleaning later

FoleyLab committed:
added a method to build 2mu in so

This is a step towards computing the DSE using pCQED

FoleyLab committed:
added build for MuMu matrix in CI basis

This doesn't break anything but it hasn't been tested for correctness.  In particular, need to pay close attention to factors of 2 and 1/2 that arise in the cross terms in mu_x mu_y, q_xy, etc. and in the overall DSE 1/2 ( lambda \cdot mu)^2

FoleyLab committed:
Nam's implementation of diis in cqed-rhf

FoleyLab committed:
Some comparisons made for LiH

FoleyLab committed:
Update pCQED-LiH-Huo-6311gss.ipynb

FoleyLab committed:
Update pCQED-LiH-Huo-6311gss.ipynb

FoleyLab committed:
updated notebook

FoleyLab committed:
updated helper_PFCI

dmejiar committed:
Extend to arbitrary number of photons

FoleyLab committed:
Merge pull request #9 from dmejiar/canonical_mo_option

Extend to arbitrary number of photons

FoleyLab committed:
updated notebooks

FoleyLab committed:
added some fcidump testing

FoleyLab committed:
updated

FoleyLab committed:
added routine to compute 1-e and 2-e energy from FCI wavefunctions

FoleyLab committed:
added h2o2p test
